develop/dev-guide-choose-driver-or-orm.md

```diff
@@ -88,7 +85,7 @@ TiDB 与 MySQL 有差异：
 <dependency>
     <groupId>mysql</groupId>
     <artifactId>mysql-connector-java</artifactId>
-    <version>8.0.28</version>
+    <version>5.1.49</version>
```

Comment:
@Icemap Please modify the version in `develop/dev-guide-optimistic-and-pessimistic-transaction.md` L69 to `5.1.49`.

Reply:
This dependency will affect the code. I will update it on another PR later.